homework/index.js

```diff
@@ -16,31 +30,116 @@
     xhr.send();
   }
 
-  function createAndAppend(name, parent, options = {}) {
-    const elem = document.createElement(name);
-    parent.appendChild(elem);
-    Object.keys(options).forEach(key => {
-      const value = options[key];
-      if (key === 'text') {
-        elem.textContent = value;
-      } else {
-        elem.setAttribute(key, value);
+  function loader(parent) {
+    const loaderDiv = createAndAppend('div', parent, { class: 'loaderDiv' });
+    createAndAppend('div', loaderDiv, { class: 'loader' });
+  }
+
+  function renderError(error) {
+    const root = document.getElementById('root');
+    createAndAppend('h1', root, { text: error.message });
+  }
+
+  function addRow(tableBody, description, className, elementContent) {
+    const tableRow = createAndAppend('tr', tableBody, { class: 'flex-div' });
+    createAndAppend('td', tableRow, {
+      text: description,
+      class: className,
+    });
+    createAndAppend('td', tableRow, {
+      text: elementContent,
+    });
+    return tableRow;
+  }
+
+  function renderRepository(tableDiv, repository) {
+    tableDiv.innerHTML = '';
+
+    const table = createAndAppend('table', tableDiv);
+    const tbody = createAndAppend('tbody', table);
+    const firstRow = addRow(tbody, 'Repository:', 'td-header', ' ');
+    createAndAppend('a', firstRow.lastChild, {
+      text: repository.name,
+      href: repository.html_url,
+      target: '_blank',
+    });
+    addRow(tbody, 'Description:', 'description-td td-header', repository.description);
+    addRow(tbody, 'Forks:', 'td-header', repository.forks_count);
+    addRow(tbody, 'Updated:', 'td-header', new Date(repository.updated_at).toLocaleString());
+  }
+
+  function renderContributions(url, contributesDiv) {
+    fetchJSON(url, (err, contributors) => {
+      if (err) {
+        renderError(err);
+        return;
       }
+
+      contributesDiv.innerHTML = '';
+      createAndAppend('h1', contributesDiv, { class: 'contributes-header', text: 'Contributions' });
+
+      const ul = createAndAppend('ul', contributesDiv);
+
+      Object.keys(contributors).forEach(contributor => {
+        const hyperlink = createAndAppend('a', ul, {
```

Comment:
An `ul` element can only have `li` elements as its children. The `a` tag should be a child of the `li`, not the other way around. And then, all elements that now have the `li` as their parent should have the `a` tag as their parent.